src/main/groovy/io/seqera/wave/service/blob/impl/BlobCacheServiceImpl.groovy

```diff
@@ -209,9 +233,41 @@ class BlobCacheServiceImpl implements BlobCacheService {
      * @return The HTTP URI from the cached layer blob is going to be downloaded
      */
     protected String blobDownloadUri(RoutePath route) {
-        StringUtils.pathConcat(blobConfig.baseUrl, route.targetPath)
+        def url = createPresignedGetUrl(blobConfig.storageBucketName, route.targetPath)
+        def downloadUrl
+
+        if( blobConfig.baseUrl ){
+            downloadUrl = url.replace("https://${blobConfig.storageBucketName}.s3.${blobConfig.storageRegion.toUpperCase()}.amazonaws.com",blobConfig.baseUrl)
+            log.info "Download URL with base url: $downloadUrl"
+        }else{
+            downloadUrl = url
+        }
+
+        log.info "Download URL: $downloadUrl"
+        return downloadUrl
```

Comment:
Please add some unit tests. Also, it would be useful to start using [Local stack](https://www.localstack.cloud/) via [Testcontainer](https://java.testcontainers.org/modules/localstack/) for more reliable testing

Reply:
ok sure

Reply:
Added unit tests